src/lib.rs

```diff
@@ -238,17 +240,24 @@ impl FuzzyFinder {
     /// let finder = FuzzyFinder::new();
     /// assert_eq!("Asia/Shanghai", finder.get_tz_name(116.3883, 39.9289));
     /// ```
+    ///
+    /// # Panics
+    ///
+    /// - Panics if `lng` or `lat` is out of range.
+    /// - Panics if `lng` or `lat` is not a number.
+    #[must_use]
     pub fn get_tz_name(&self, lng: f64, lat: f64) -> &str {
         for zoom in self.min_zoom..self.max_zoom {
             let idx = deg2num(lng, lat, zoom);
             let k = &(idx.0, idx.1, zoom);
-            let ret = self.all.get(&k);
+            let ret = self.all.get(k);
             if ret.is_none() {
                 continue;
             }
-            return ret.unwrap();
+            // Be explicit about crashing if we get a None value.
+            return ret.expect("Yikes! Got a None value for the TZ name. That shouldn't happen.");
         }
-        return "";
+        ""
```

Comment:
Return `""` is intended, since FuzzyFinder not aims to cover 100% areas on earth:

![](https://user-images.githubusercontent.com/13536789/200174943-7d40661e-bda5-4b79-a867-ec637e245a49.png)

It will be more easy for down stream packages like Ruby/Python binding to handle this.

Reply:
That's fine. But Rust's idiomatic code is to not use "return" when returning at the end of a function. This does the same thing- it returns an empty string. It's just a more idiomatic way of expressing the same thing.